sequencer/src/block2/queryable.rs

```diff
@@ -46,15 +49,17 @@ impl QueryablePayload for Payload {
         // sum over these tx table lens
         let mut result = 0;
         for &offset in ns_end_offsets.iter().take(ns_end_offsets.len() - 1) {
-            let tx_table_len = get_table_len(&self.payload, offset).try_into().unwrap_or(0);
+            let tx_table = TxTable::<u32>::from_bytes(&self.raw_payload);
```

Comment:
The `TxTable` instantiation has been [moved out of the loop](7e1a1ee). I agree that it would be good to avoid cloning the payload and instead storing only the necessary information for answering queries later. I am not comfortable to do this now as we don't have unit tests yet for TxTable, yet I added a TODO in 0095537. WDYT?

Reply:
After a closer look I realized that the stateful version of the transactions table is useful only for testing purposes. So I introduced as you suggested the stateless version of it to use in non-test code. See 3cc5b30.